dir.props

```diff
@@ -186,16 +192,16 @@
   <PropertyGroup Condition="'$(NetCoreSurface)' != 'true'">
     <DefineConstants>$(DefineConstants);FEATURE_APARTMENT_STATE</DefineConstants>
     <DefineConstants>$(DefineConstants);FEATURE_APM</DefineConstants>
-    <DefineConstants>$(DefineConstants);FEATURE_APPDOMAIN</DefineConstants>
-    <FeatureAppDomain>true</FeatureAppDomain>
+    <DefineConstants Condition="'$(MonoBuild)' != 'true'">$(DefineConstants);FEATURE_APPDOMAIN</DefineConstants>
+    <FeatureAppDomain Condition="'$(MonoBuild)' != 'true'">true</FeatureAppDomain>
```

Comment:
I did not think there are enough of the Mono-specific entries. Thoughts welcome.